walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -34,38 +34,46 @@
     }
 
     // Seperate relevant TimeRanges from events, put into ArrayList and sort by ascending meeting start time
-    List<TimeRange> meetings = new ArrayList<>();
-    for (Event event : events) {
+    List<TimeRange> attendedMeetings = new ArrayList<>();
+    for (Event event: events) {
       // First check if the event in question contains people from request
       Set<String> attendees = new HashSet<>(request.getAttendees());
       attendees.retainAll(event.getAttendees());
 
-      int numMeetings = meetings.size();
+      if (!attendees.isEmpty()){
+        attendedMeetings.add(event.getWhen());
+      }
+    }
+
+    Collections.sort(attendedMeetings, TimeRange.ORDER_BY_START);
+
+    List<TimeRange> validMeetings = new ArrayList<>();
+
+    for (TimeRange meeting : attendedMeetings) {
+      
+      int numMeetings = validMeetings.size();
 
       // Only add meeting if A) there are people attending and B) it's not nested within the previous meeting
-      if(!attendees.isEmpty() && 
-        (numMeetings == 0 || event.getWhen().end() > meetings.get(numMeetings - 1).end())) {
-        meetings.add(event.getWhen());
+      if(numMeetings == 0 
+        || meeting.end() > validMeetings.get(numMeetings - 1).end()) {
+        validMeetings.add(meeting);
```

Comment:
What happens if two or more meetings start at the same time but end with different times?

Reply:
That's accounted for! If they're not completely nested, they're still counted as valid meetings. However, findNextTime would return null.

Reply:
This is my understanding. Please let me know if I'm wrong:
If the input meetings are [3,4] and [3, 5], it will add both of them.
If the input meetings are [3,4], [3,7], [3,5], [3,6], it will add [3,4], [3,7], [3,6].
Are these ok with our findNextTime?

Reply:
It would add [3,4], [3,7] because, while it's iterating over [3,5], [3,6], it would check back to the last added valid meeting, which is [3,7] in both cases. Because these meetings end before [3,7] does, they would never be added. Then once it gets passed to findNextTime, [3,4] would return null!

Reply:
Oh. You are right. This seems to be a complicated enough logic that deserves more commenting. Can you add something like:
"// Fill in validMeetings. The resulting validaMeetings will have the meetings in ascending order (in both starting and end time). If two or more meetings in attendedMeetings start at the same time, one or more of those meetings will be chosen with the longest of them will always be chosen."

Reply:
Done!